src/AI/Muriel.java

```diff
@@ -955,20 +960,27 @@ private void queueUnitProductionActions(GameMap gameMap)
     } // ~while( still choosing units to build )
 
     // Build infantry from any remaining facilities.
-    UnitModel infModel = myCo.getUnitModel(UnitModel.TROOP);
-    final int infCost = myCo.getCost(infModel);
-    while( (budget >= infCost) && (CPI.availableUnitModels.contains(infModel)) )
+    UnitModel infModel = myArmy.cos[0].getUnitModel(UnitModel.TROOP);
```

Comment:
Isn't this incorrect when `myArmy` has multiple COs?

Also on line 808.

Reply:
No. This is just "`Army` doesn't have a copy of the model list directly on hand for querying".
The code on the following lines has to cope with the CO-agnostic nature of `UnitModel`; this instance by just asking for the price directly, 808 via `getAverageCostFor()`.

Reply:
On the note of "making the unit type list more available", would this be a case where it makes sense to extend a collection class?
A list of units *is-a* list of things, but it also wants to provide non-standard query APIs that only make sense in this context.

An alternative would be to move the implementations of the filter functions to static methods and have the `Commander` copies just call those.

Reply:
> No. This is just "Army doesn't have a copy of the model list directly on hand for querying".

Ah. Well if `UnitModel`s aren't CO-specific anymore, it might make sense to have some way of obtaining them without going through a CO. Both for convenience and to clarify intent. That can probably be future cleanup though.

Re: extending a collection class - maybe? Not sure what you have in mind here in terms of API.

Reply:
The API would be:
Commander's unit list and anywhere else it's cached would now be our new bespoke class.
That class would own the `getAllModels(long role)` family of functions, with Commander's versions just dispatching to the collection (to avoid code churn).

...I think that's it?

Reply:
I'm not sure how that's any better than what we have. But maybe I'm still not picking up what you're putting down.

Reply:
It's not a better API, it's allowing our same query API to exist in other places than just `Commander`.
So I could go to the `GameRules` and say "Give me the first defined `TROOP`".